### PR TITLE
Release notes for 2.0.0

### PR DESCRIPTION
**Pull request by jumaffre**

### CODE REVIEW

File: release_notes_2.0.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Release notes 2.0
```
Comment (jumaffre) [inline release_notes_2.0.md:1]: To be moved to `CHANGELOG.md` when ready.

File: release_notes_2.0.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+TODO:
+
+- [ ] Open Enclave?
+- [ ] TLS?
+- [ ] Commit evidence?
```
Comment (eddyashton) [inline release_notes_2.0.md:7]: TODO: Remove before merging.

File: release_notes_2.0.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+- Added `get_untrusted_host_time_v1` API. This can be used to retrieve a timestamp during endpoint execution, accurate to within a few milliseconds. Note that this timestamp comes directly from the host so is not trusted, and should not be used to make sensitive decisions within a transaction (#2550).
+- Added `get_quotes_for_all_trusted_nodes_v1` API. This returns the ID and quote for all nodes which are currently trusted and participating in the service, for live audit (#2511).
+- Added `get_metrics_v1` API to `BaseEndpointRegistry` for applications that do not make use of builtins and want to version or customise metrics output.
+- `ccf::historical::adapter_v2` now returns 404, with either `TransactionPendingOrUnknown` or `TransactionInvalid`, rather than 400 when a user performs a historical query for a transaction id that is not committed.
+- `ccf::historical::AbstractStateCache::drop_requests()` renamed to `drop_cached_states()` (#3187).
+- `get_state_at()` now returns receipts for signature transactions (#2785), see [documentation](https://microsoft.github.io/CCF/main/use_apps/verify_tx.html#transaction-receipts) for details.
+
```
Comment (eddyashton) [inline release_notes_2.0.md:24]:
```suggestion
- Added indexing system to speed up historical queries (#3280, #3444).
```

File: release_notes_2.0.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+- Fixed issue with ledger inconsistency when starting a new joiner node without a snapshot but with an existing ledger prefix (#3064).
+- Fixed issue with join nodes which could get stuck if an election was triggered while catching up (#3169).
+
+---
+
+## Client API
+
+- Added `GET /gov/jwt_keys/all` endpoint (#2519).
+- Added new operator RPC `GET /node/js_metrics` returning the JavaScript bytecode size and whether the bytecode is used (#2643).
+- Added a new `GET /node/metrics` endpoint which includes the count of active and peak concurrent sessions handled by the node (#2596).
```
Comment (rengolin) [inline release_notes_2.0.md:100]: Added `GET /node/jwt_metrics`

File: release_notes_2.0.md (hunk 2 above)
Comment (rengolin) [reply release_notes_2.0.md:7]: TLS:
 * Removed MbedTLS as TLS/Crypto library provider
 * Added OpenSSL as TLS provider
 * Supports TLS 1.3 (default) but also 1.2 (if forced)

File: release_notes_2.0.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+- Added endpoint to obtain service configuration via `GET /node/service/configuration` (#3251).
+- Added QuickJS version to RPC `GET /node/version` (#2643).
+
+- Schema of `GET /network/nodes/{node_id}` and `GET /network/nodes` endpoints has been modified to include all RPC interfaces (#3300).
+- Improved performance for lookup of path-templated endpoints (#2918).
+- CCF now responds to HTTP requests that could not be parsed with a 400 response including error details (#2652).
+
+- Websockets endpoints are no longer supported. Usage is insufficient to justify ongoing maintenance.
+- The `ccf` Python package no longer provides utilities to issue requests to a running CCF service. This is because CCF supports widely-used client-server protocols (TLS, HTTP) that should already be provided by libraries for all programming languages. The `ccf` Python package can still be used to audit the ledger and snapshot files (#3386).
+
```
Comment (eddyashton) [inline release_notes_2.0.md:114]: I think that under this division, the TLS changes go under Client API? So:
```suggestion
- Added OpenSSL as TLS provider, replacing MbedTLS
- Added support for TLS 1.3


```

File: release_notes_2.0.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+- [ ] Commit evidence?
+
+---
+
+## Developer API
+
+### C++
+
+- CCF is now built with Clang 10. It is recommended that C++ applications upgrade to Clang 10 as well.
```
Comment (achamayou) [inline release_notes_2.0.md:13]:
```suggestion
- CCF is now built with Clang 10. It is strongly recommended that C++ applications upgrade to Clang 10 as well.
```

File: release_notes_2.0.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+- Raised the minimum supported CMake version for building CCF to 3.16 (#2946).
+- Removed `mbedtls` as cryptography library.
```
Comment (achamayou) [inline release_notes_2.0.md:15]:
```suggestion
- Removed `mbedtls` as cryptography and TLS library.
```

File: release_notes_2.0.md (hunk 3 above)
Comment (achamayou) [reply release_notes_2.0.md:25]: A link to the doc page would be really good "See documentation for details".

File: release_notes_2.0.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+- `ccf.crypto.verifySignature()` previously required DER-encoded ECDSA signatures and now requires IEEE P1363 encoded signatures, aligning with the behavior of the Web Crypto API (#2735).
+- `ccf.historical.getStateRange` / `ccf.historical.dropCachedStates` JavaScript APIs to manually retrieve historical state in endpoints declared as `"mode": "readonly"` (#3033).
+- JavaScript endpoints with `"mode": "historical"` now expose the historical KV at `ccf.historicalState.kv` while `ccf.kv` always refers to the current KV state. Applications relying on the old behaviour should make their code forward-compatible before upgrading to 2.x with `const kv = ccf.historicalState.kv || ccf.kv`.
+- Receipts accessible through JavaScript no longer contain the redundant `root` hash field. Applications should be changed to not rely on this field anymore before upgrading to 2.x.
+
+---
+
+## Governance
+
+- Updated `actions.js` constitution fragment to record service-endorsed node certificate on the `transition_node_to_trusted` action. The constitution should be updated using the existing `set_constitution` proposal (#2844).
```
Comment (achamayou) [inline release_notes_2.0.md:46]:
```suggestion
- Updated `actions.js` constitution fragment to record service-endorsed node certificate on the `transition_node_to_trusted` action. The constitution must be updated using the existing `set_constitution` proposal (#2844).
```

File: release_notes_2.0.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+### Certificate(s) Validity Period
+
+- Nodes certificates validity period is no longer hardcoded and can instead be set by operators and renewed by members (#2924):
+
+  - The new `node_certificate.initial_validity_days` (defaults to 1 day) configuration entry lets operators set the initial validity period for the node certificate (valid from the current system time).
+  - The new `command.start.service_configuration.maximum_node_certificate_validity_days` (defaults to 365 days) configuration entry sets the maximum validity period allowed for node certificates.
+  - The new `set_node_certificate_validity` proposal action allows members to renew a node certificate (or `set_all_nodes_certificate_validity` equivalent action to renew _all_ trusted nodes certificates).
+
+- Service certificate validity period is no longer hardcoded and can instead be set by operators and renewed by members (#3363):
```
Comment (achamayou) [inline release_notes_2.0.md:70]:
```suggestion
- Service certificate validity period is no longer hardcoded and must instead be set by operators and renewed by members (#3363):
```

File: release_notes_2.0.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+- Raised the minimum supported CMake version for building CCF to 3.16 (#2946).
+- Removed `mbedtls` as cryptography and TLS library.
+
+- Added `get_untrusted_host_time_v1` API. This can be used to retrieve a timestamp during endpoint execution, accurate to within a few milliseconds. Note that this timestamp comes directly from the host so is not trusted, and should not be used to make sensitive decisions within a transaction (#2550).
+- Added `get_quotes_for_all_trusted_nodes_v1` API. This returns the ID and quote for all nodes which are currently trusted and participating in the service, for live audit (#2511).
+- Added `get_metrics_v1` API to `BaseEndpointRegistry` for applications that do not make use of builtins and want to version or customise metrics output.
+- Added `set_claims_digest()` API to `RpcContext`, see [documentation](https://microsoft.github.io/CCF/main/build_apps/logging_cpp.html#user-defined-claims-in-receipts) on how to use it to attach application-defined claims to transaction receipts.
+- Added indexing system to speed up historical queries (#3280, #3444).
```
Comment (achamayou) [inline release_notes_2.0.md:23]:
```suggestion
- Added [indexing system](https://microsoft.github.io/CCF/main/architecture/indexing.html) to speed up historical queries (#3280, #3444).
```

File: release_notes_2.0.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+---
+
+## Auditor
+
+- Receipts now include the endorsed certificate of the node, as well as its node id, for convenience (#2991).
+- Retired nodes are now removed from the store/ledger as soon as their retirement is committed (#3409).
+- Service-endorsed node certificates are now recorded in a new `public:ccf.gov.nodes.endorsed_certificates` table, while the existing `cert` field in the `public:ccf.gov.nodes.info` table is now deprecated (#2844).
+- New `split_ledger.py` utility to split existing ledger files (#3129).
+- Python `ccf.read_ledger` module now accepts custom formatting rules for the key and value based on the key-value store table name (#2791).
```
Comment (achamayou) [inline release_notes_2.0.md:107]: - [Ledger entries](https://microsoft.github.io/CCF/main/architecture/ledger.html#transaction-format) now contain a `commit_evidence_digest`, as well as an optional `claims_digest` when `set_claims_digest()` is used. The digest of the write set was previously the per-transaction leaf in the Merkle Tree, but is now combined with the digest of the commit evidence and optionally the user claims when present. [Receipt verification instructions](https://microsoft.github.io/CCF/main/audit/receipts.html) have been amended accordingly.

File: release_notes_2.0.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
+# Release notes 2.0
+
+TODO:
+
+- [ ] Commit evidence?
```
Comment (achamayou) [inline release_notes_2.0.md:5]:
```suggestion
```

**Review by achamayou — APPROVED** [review]

**Pull request merged**